src/evm/StateMachine.ts

```diff
@@ -0,0 +1,164 @@
+import { ethers, BytesLike, hexlify } from "ethers";
+import { TransactionStruct } from "@typechain-types/contracts/V1/DataTypes";
+import { ContractExecuter } from "./ContractExecuter";
+import { EVM, Log } from "@ethereumjs/evm";
+import { Address } from "@ethereumjs/util"
+/**
+ * Provides a typed interface to interact with a state machine contract
+ * running in an EVM environment
+ */
+export class StateMachine {
+    readonly contractExecuter: ContractExecuter;
+    readonly contractInterface: ethers.Interface;
+
+
+    /**
+     * Creates a new StateMachine instance
+     * @param contractExecuter The contract executer
+     * @param contractInterface The contract interface
+     */
+    constructor(
+        contractExecuter: ContractExecuter,
+        contractInterface: ethers.Interface
+    ) {
+        this.contractExecuter = contractExecuter;
+        this.contractInterface = contractInterface;
+    }
+
+    private callData(functionName: string, args: any[] = []): Uint8Array {
+        return ethers.getBytes(this.contractInterface.encodeFunctionData(functionName, args));
```

Comment:
This function makes sense - would name it getEncodedCalldata

Reply:
done

Reply:
renames to StateChannelSession